lib/galaxy/tools/parameters/basic.py

```diff
@@ -1361,9 +1361,12 @@ def get_legal_values(self, trans, other_values):
         legal_values = self.get_column_list(trans, other_values)
 
         value = other_values.get(self.name)
-        if value is not None and value not in legal_values and self.is_file_empty(trans, other_values):
-            value = value if isinstance(value, list) else [value]
-            legal_values.extend(value)
+        if value is not None:
+            # There are cases where 'value' is a string of comma separated values. This ensures
+            # that it is converted into a list, with extra whitespace around items removed.
+            value = util.listify(value, do_strip=True)
```

Comment:
is this a failure of the calling function to listify the values?
I feel like the tool input should be listified much earlier on

Reply:
I think the issue is actually here: https://github.com/galaxyproject/galaxy/blob/84647cbd0a05b8dd87a5ffe4662c033ad050b0ee/lib/galaxy/tools/parameters/basic.py#L1256

Reply:
I'd rather not make that assumption about input; processing this collection of items as a list is just the method's implementation detail, which may or may not change in the future. Besides, accepting a list or a tuple, or a comma-separated string seems more flexible.

Reply:
The issue seems to be that the logic to convert the column spec to a list is very different than what the listify function does.
The column list spec seems to allow multiline and numbers prefixed with a 'c'.